StRoot/PWGTools/BadRunQA/README.md

```diff
@@ -1,78 +1,97 @@
-2. Install the gnuplot (version older than 5.0)
-3. Change the "/star/u/jdb/.exodus/bin/gnuplot <<EOF" in to "gnuplot <<EOF" in "plot_runbyrun.sh"
-4. Run the same command to do the run-by-run QA: $> bash runbyrun_v2.sh FILENAME [e.g qahist.root ] NAMELIST [e.g name.list ] SYSTEM [e.g Au+Au ] ENERGY [19.6]
-
-##############################################
-##########   CHANGE THE PLOTTING     #########
-##############################################
-1. If you only want to plot, prepare the data file with the same formate as runInfo_runidvsavgdedx.txt. Then run: bash plot_runbyrun.sh HISTNAME [e.g runidvsavgdedx] SYSTEM [e.g Au+Au] ENERGY [19.6] NSIGMA [e.g 20], for example:
-      $> bash plot_runbyrun.sh runidvsavgdedx Au+Au 19.6 20
-The default NSIGMA (Y-range) is 20-RMS for the good runs
-2. If you want to change the color:
-Change "color1"..."color6" with whatever the color you prefer in "plot_runbyrun.sh". Color is in hex mode, see https://www.colorhexa.com/ for colors.
-3. If you want to change the output file format, change '.eps' to any image format (e.g '.jpg') in "plot_runbyrun.sh"
-
-####################################################
-##########   CHANGING THE INPUT BETA     ###########
-## internal segmentation identification parameter ##
-####################################################
-The current package has an automated searching function to search for the input parameters using the segmentation identification algorithm (input to "Segmentation2.py" file). You don't need to make any changes in principle. But the package still gives you the freedom to tune the input parameters. The current setup will loop "0.5, 1, 2, 5, 9" these 5 beta values and find one that gives you the most badruns. But you can vary the beta value if you want. 
-In short, the number of segmentation and the number of bad runs are correlated. The different beta will give you a different number of segmentations, but should still give you a similar number of badruns.
+## Introduction
+
+This software is designed to segment time series data points using a change point detection algorithm. The algorithm detects changes in the mean and variance of the data points and segments the data into different segments based on these changes. The software then removes outliers in each segment by removing points that are beyond a certain factor (default is 5) standard deviations from the mean of each segment. 
+The code will iteration on both its change point detection and outlier rejection until no more outliers are discovered.
+
+This code works on both Linux and Windows. It should work in Mac but it has not been tested.
+
+## Installation
+
+To install the software, follow these steps:
+The EASY way (Linux only. For Windows and Mac, use the recommended way):
+1. Run the following command: `python3 -m pip install --user matplotlib ruptures pandas numpy scipy pyfiglet uproot scikit-learn`
```

Comment:
Usually it helps to freeze the dependencies in `requirements.txt` (or in `pyproject.toml`). It should help with reproducibility of results as well.

Reply:
I have updated my repo to include requirements.txt for Linux. For some reasons the packages must be installed with anaconda on Windows so I don't have one for Windows. 

I have also updated my run-by-run code to include a legacy mode to emulate the behavior of the old version (v2).

Reply:
I tried to use requirements.txt on my Windows desktop and Windows laptop with pip3 install -r requirements.txt and the package "rupture" gives me errors. Something along the line of "error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools"". I did update my Microsoft C++ Build Tools but it's still not working. I can still get that particular package to work on Windows, but only if I install with conda, i.e. conda install ruptures. 

However, this requirements.txt definitely works on RCAS.